From c1a38f1b8b4a209ec970d8929450f88f2234c0e7 Mon Sep 17 00:00:00 2001
From: Artem Senichev <a.senichev@yadro.com>
Date: Thu, 21 Feb 2019 15:15:46 +0300
Subject: [PATCH] HTMGT: Handle 'invalid' status of DCMI messages

OpenBMC responses with status INVALID (0xC1) for power limit IPMI requests:
https://github.com/openbmc/phosphor-host-ipmid/commit/2c2af2ca713f56b117ad2c8c1a1f0e370b7c2d9c
Hostboot doesn't handle that status properly, and, as a result, register an
error and stop OCC initialization procedure.

According to section 6.6 Power management of DCMI specification, if power
management feature is disabled, the related commands should be unsupported.

Signed-off-by: Artem Senichev <a.senichev@yadro.com>
---
 src/usr/ipmiext/ipmidcmi.C | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/usr/ipmiext/ipmidcmi.C b/src/usr/ipmiext/ipmidcmi.C
index a1b752baf..4cfd61aee 100644
--- a/src/usr/ipmiext/ipmidcmi.C
+++ b/src/usr/ipmiext/ipmidcmi.C
@@ -71,8 +71,12 @@ namespace SENSOR
         if( l_err == NULL )
         {
             // make sure the completion code is good, then read the bytes
-            if( (l_cc == POWER_LIMIT_NOT_ACTIVE) ||
-                    (l_cc == POWER_LIMIT_ACTIVE) )
+            if( cc == IPMI::CC_INVALID )
+            {
+                TRACFCOMP(g_trac_ipmi, "Invalid response on power limit");
+            }
+            else if( (l_cc == POWER_LIMIT_NOT_ACTIVE) ||
+                     (l_cc == POWER_LIMIT_ACTIVE) )
             {
                 // from the dcmi spec
                 // byte 1   completion code
-- 
2.21.0

